File: sample-music-backend/src/main/java/com/sample/music/service/PublicityService.java
package com.sample.music.service;

import com.github.pagehelper.PageHelper;
import com.sample.music.mapper.PublicityMapper;
import com.sample.music.pojo.dto.PageBean;
import com.sample.music.pojo.entity.Publicity;
import com.sample.music.pojo.vo.view.PublicityView;
import com.sample.music.utils.ThreadLocalUtil;
import com.sample.music.utils.UserContext;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Map;

@Service
@RequiredArgsConstructor
public class PublicityService {
    private final PublicityMapper publicityMapper;
    private final FileManageService fileManageService;

    public void insertPublicity(Publicity publicity) {
        if(publicity.getImageFiles() != null) {
            String avatarUrl = fileManageService.uploadFile(publicity.getImageFiles(), "image");
            publicity.setCover(avatarUrl);
        }else {
            publicity.setCover(null);
        }
        Long userId = UserContext.getUser();
        publicity.setUserId(userId);
        publicity.setViewers(0);
        publicityMapper.insertPublicity(publicity);
    }

    public PageBean<PublicityView> conditionAndPagedQuery(String condition, String publicityType, Integer pageNum, Integer pageSize) {
        PageBean<PublicityView> pb = new PageBean<>();
        PageHelper.startPage(pageNum, pageSize);
        List<PublicityView> publicityViews;// 直接使用分页结果的size作为总数
        if (condition == null || condition.isEmpty()) {
            // 没有条件，执行分页查询
            publicityViews = publicityMapper.PagedQuery(publicityType);
        } else {
            // 有条件，执行条件查询
            publicityViews = publicityMapper.conditionAndPagedQuery(condition, publicityType);
        }
        pb.setTotal(publicityViews.size()); // 直接使用分页结果的size作为总数
        pb.setItems(publicityViews);
        return pb;
    }

    public PublicityView query(Long id) {
        PublicityView query = publicityMapper.query(id);
        return query;
    }
}

File: sample-music-frontend/src/api/service/PublicityService.js
import service from "@/api/request";

const PublicityService = {
  insertPublicity: function (publicity) {
    return service({
      method: 'POST',
      url: '/api/publicity/insert',
      data: publicity
    }).then(response => {
      return response;
    }).catch(error => {
      return error;
    })
  },
  conditionAndPaged: function (condition, publicityType, pageNum, pageSize) {
    // 创建一个URLSearchParams对象
    const params = new URLSearchParams();

    // 检查condition是否存在，如果存在则添加到查询参数中
    if (condition) {
      params.append('condition', condition);
    }

    // 检查publicityType是否存在，如果存在则添加到查询参数中
    if (publicityType) {
      params.append('publicityType', publicityType);
    }

    // 检查pageNum是否存在，如果存在则添加到查询参数中
    if (pageNum) {
      params.append('pageNum', pageNum);
    }

    // 检查pageSize是否存在，如果存在则添加到查询参数中
    if (pageSize) {
      params.append('pageSize', pageSize);
    }

    // 构建完整的URL
    const url = `/api/publicity/query/conditionAndPaged?${params}`;
    return service({
      method: "GET",
      url: url,
    }).then(response => {
      return response;
    }).catch(error => {
      throw error;
    })
  },

  query: function (id) {
    // 构建完整的URL
    const url = `/api/publicity/query/${id}`;
    return service({
      method: "GET",
      url: url,
    }).then(response => {
      return response;
    }).catch(error => {
      throw error;
    })
  }
}
export default PublicityService;